tests/compatibility-test.py

```diff
@@ -151,9 +151,9 @@ def test_ray_serve(self):
             self.fail(f"Fail to execute test_ray_serve_2.py. The exit code is {exit_code}.")
 
     @unittest.skipIf(
-        ray_version == '2.8.0' or ray_version == 'nightly',
-        'test_detached_actor is too flaky with Ray 2.8.0 and Ray nightly.'
-        'Therefore, skip it until https://github.com/ray-project/ray/issues/41343 is solved.'
```

Comment:
This is the test that is unskipped in this PR.

Reply:
Why does [Compatibility Test 2.8.0](https://github.com/ray-project/kuberay/actions/runs/7332768734/job/19967438402?pr=1770) still fail with `test_detached_actor`? It should be skipped.

Reply:
Not sure how this is possible, still investigating. According to the code it looks like we are still skipping it.

Reply:
I suspect it's because the decorators are evaluated when the function is defined, not at the first the time the function is executed. So when it's evaluated, the global variable `ray_version` is still set to its initial value of `2.9.0`, so it doesn't get skipped.  

Before this PR the initial value was `2.8.0`, so the test was actually being skipped for all compatibility tests (2.5, 2.6, 2.7, 2.8, nightly). I'll just fix it directly in this PR.

Reply:
Fixed! 3b0718d